### ISSUE TITLE
Allow arrays as input to time-varying properties

### ISSUE THREAD

**Issue by bryanwweber** [Cantera/enhancements#17]
**Abstract**

Allow two-column arrays to be used to specify time-varying properties of the `Reactor` class.

**Motivation**

Currently, we have to use a function to calculate things like velocity, etc. that can vary with time. (I'm thinking here mostly of the Python interface but the idea may be appropriate for others as well.) This requires calling out to Python on every time step. It might be faster to pass a two-column array with time and values and have this done in C++ for cases where this is possible. Certainly, adding this kind of capability would make it much easier for users to specify this kind of data, but I'm not sure how common this use case is. 

This is inspired by the need for RCM simulations to specify pre-calculated volume (i.e., velocity) as a function of time. Of course, this should be benchmarked for speed to see if there is actually any improvement.

**Possible Solutions**

The way this could work from Python is something like:
```python
import numpy as np
vel_array = np.genfromtxt('velocity-trace.txt')
# vel_array looks like [[0.0, 1.0], [1.0, 2.0], ...]
# Normal Cantera Reactor or IdealGasReactor setup
reac = ct.IdealGasReactor(gas)
w = ct.Wall(reac, env, velocity=vel_array)
# Rest of the ReactorNetwork simulation
```

Something like this would also work for the heat transfer function for `Reactor`, as well as the mass flow rate for flow devices (and possibly other cases I'm missing).

One question is how should time values between values actually specified should be handled. In other words, given `vel_array = [[0.0, 1.0], [1.0, 2.0]]`, if `t = 0.5`, what velocity should be used? The implementation of this in CHEMKIN uses the velocity from the nearest previous time value, i.e., for `t = 0.5`, `v = 1.0`.

Other possible ideas would be to fit a function (linear, quadratic, etc.) between the data points and interpolate. I'm not in favor of this idea, since we don't know what type of function should be used, and if a user wants something like this they can fall back to the existing `Func1`-type interface.

**References**

This Google Group post has one possible (typo-filled) implementation of a function to read a volume-vs-time curve into an array and return the appropriate velocity: https://groups.google.com/forum/#!topic/cantera-users/5yoAUvAdOLs

Another, class-based, implementation of the same idea is here: https://github.com/bryanwweber/CanSen/blob/43ffe2f2f6959d9208a4a44d508866e029eb567b/cansen/profiles.py#L7-L59

**Comment by ischoegl** [comment]
I implemented a version of this as part of a more comprehensive `Cylinder`/`Piston`/`Solenoid` combination, which was an illustration for Cantera/cantera#632. At the time, I decided to remove it after feedback that this is better placed in an applications-oriented derivative package (files are parked in ischoegl/ctapp@55feb5fd93dc457c0df33278aafa1afe9a54c0c0, specifically the `DrivenPiston` variant of a `Wall` object). 

The alternative suggested here is much less intrusive, i.e. provide for an alternative to `Func1` for various time-varying inputs. I simply wanted to mention the option of having dedicated wall objects, although I agree that there is a risk of introducing 'bloat' due to overly specific implementations.
